build-support/bin/release.sh

```diff
@@ -7,12 +7,64 @@ set -e
 ROOT=$(cd $(dirname "${BASH_SOURCE[0]}") && cd "$(git rev-parse --show-toplevel)" && pwd)
 source ${ROOT}/build-support/common.sh
 
-PY=$(which python2.7 || exit 0)
-[[ -n "${PY}" ]] || die "You must have python2.7 installed and on the path to release."
+# Note we define all options here, but only parse some at the top of the script and parse the rest
```

Comment:
While it does look like the double options parsing is necessary (according to https://stackoverflow.com/questions/15079009/is-it-possible-to-specify-the-order-getopts-conditions-are-executed), it doesn't look like it's necessary for the two parses to be separated within the file. I think it would be preferable to have the two while loops right after one another.

The consumption of the variables defined (`py_version_number` in particular) happens lazily in functions... so it's fine if they're not set until later in the file.